doc/subsystems/logging/logger.rst

```diff
@@ -207,15 +229,36 @@ In order to use instance level filtering following steps must be performed:
 Note that when logger is disabled logger instance and pointer to that instance
 are not created.
 
-- logger can be used in function
+In order to use instance logging API in the source file a compile-time log level
+must be set using :c:macro:`LOG_LEVEL_SET`. Optionally, a compile time log level
+for the module can be specified in the parameter. Default log level
+(:option:`CONFIG_LOG_DEFAULT_LEVEL`) is used if custom log level is not
+provided.
 
 .. code-block:: c
 
+   LOG_LEVEL_SET(CONFIG_FOO_LOG_LEVEL);
+
    void foo_init(foo_object *f)
    {
    	LOG_INST_INF(f->log, "Initialized.");
    }
 
+In order to use instance logging API in the header file a compile-time log level
```

Comment:
Change to:

    In order to use the instance logging API in a header file, a compile-time log level ...